### DISCUSSION TITLE
Is an avahi / bonjour domain extension always ".local"?

### DISCUSSION THREAD

**Discussion by markosjal** [alexpevzner/sane-airscan#228] [General]
Trying to add dynamic IP addressing support to one of my eSCL projects but running into a big question. As I find no other place to post such questions where they get answere, , seems I'll ask here....

Is an avahi / bonjour domain extension always ".local"?

Is ".local" just the default and it can be customized or does Avahi / Bonjour ONLY work on names with the ".local" extension?

I have also seen ".lan" , ".home", etc, but not with Avahi/Bonjour as of yet.

Anyone have a clue?

Thanks in advance

**Comment by alexpevzner** [comment]
See explanation on the page 5 of the following document: https://datatracker.ietf.org/doc/rfc6762/

In few words, addresses in the form `*.local` must resolve to the link-local IP addresses (i.e., 169.254/16 or FE80::/10) and DNS queries for `*.local` addresses must be sent to the link-local multicast address (224.0.0.251 or FF02::FB).

As packets send to or from link-local addresses never bypasses a router, link-local segments are not directly connected, so names belonging to different link-local segments are not guaranteed to be unique between segments.

In another words, if computer is connected to two local networks (say, wired and wireless), both networks may have a host under the same name, and these two hosts may be actually different. `*.local` addresses are valid only within LAN segment they belong to and every such segment is isolated from other LAN segments.

From another hand, "normal" DNS names can be made resolvable via milticast DNS. So using the `.local` top-level domain is about the range of IP addresses that belongs to such a domain and about definition scope of such addresses, not about method to resolve it (mDNS vs classical DNS)

**Comment by markosjal** [comment]
Alex,

Thanks for the link. 

I  read through it but I do not think much of it really applies directly to Bonjour/Avahi which I believe to be a subset of the mDNS protocol.
